GG/src/ListBox.cpp

```diff
@@ -1173,6 +1173,7 @@ void ListBox::BringRowIntoView(iterator it)
 
     Y y_offset(BORDER_THICK), it_y_offset(BORDER_THICK);
     Y first_row_y_offset(BORDER_THICK), last_row_y_offset(BORDER_THICK);
+    iterator it_end_row = --m_rows.end();
```

Comment:
Yes.

The line above
````
    // m_first_row_shown only equals end() if the list is empty, hence 'it' is invalid.
    if (m_first_row_shown == m_rows.end())
        return;
````
should be checking an invariant of the ListBox that is preserved in all the insert/remove operations.  If this is not true, other parts of ListBox are broken.